core/src/main/scala/cromwell/core/path/EvenBetterPathMethods.scala

```diff
@@ -101,24 +101,29 @@ trait EvenBetterPathMethods {
     write(content)(openOptions, Codec.UTF8)
   }
 
+  private def fileIoErrorPf[A]: PartialFunction[Throwable, Try[A]] = {
+    case ex: Throwable => Failure(new IOException(s"Could not read from ${this.pathAsString}: ${ex.getMessage}", ex))
+  }
+
   /*
    * The input stream will be closed when this method returns, which means the f function
    * cannot leak an open stream.
    */
   def withReader[A](f: BufferedReader => A)(implicit ec: ExecutionContext): A = {
-
     // Use an input reader to convert the byte stream to character stream. Buffered reader for efficiency.
-    tryWithResource(() => new BufferedReader(new InputStreamReader(this.mediaInputStream, Codec.UTF8.name)))(f).recoverWith({
-      case failure => Failure(new IOException(s"Could not read from ${this.pathAsString}: ${failure.getMessage}", failure))
-    }).get
+    tryWithResource(() => new BufferedReader(new InputStreamReader(this.mediaInputStream, Codec.UTF8.name)))(f).recoverWith(fileIoErrorPf).get
+  }
+
+  def withBufferedStream[A](f: BufferedInputStream => A)(implicit ec: ExecutionContext): A = {
```

Comment:
The description on this PR would be perfect as a Scaladoc here to explain the difference between the two methods. 🙏 🙂